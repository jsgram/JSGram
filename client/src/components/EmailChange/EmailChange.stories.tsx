import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { Index } from './index';

export const emailChangeState = {
    email: 'maks@gmail.com',
    user: {},
};

export const emailChangeAction = {
    setEmailText: action('setEmailText'),
    changeEmail: action('changeEmail'),
};

storiesOf('EmailChange', module)
    .add('change email', () => <Index {...{...emailChangeState, ...emailChangeAction}}/>);
