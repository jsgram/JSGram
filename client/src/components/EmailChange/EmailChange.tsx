import React from 'react';
import '../styles/style.scss';
import {Button, Form, FormProps, Input} from 'reactstrap';
import '../styles/Profile.scss';

export default class EmailChange extends React.Component<any> {

    constructor(props: FormProps) {
        super(props);
        this.onEmailChange = this.onEmailChange.bind(this);
        this.changeEmail = this.changeEmail.bind(this);
    }

    public onEmailChange(event: React.ChangeEvent<HTMLInputElement>): void {
        this.props.setEmailText(event.target.value);
    }

    public changeEmail(): void {
        this.props.changeEmail();
    }

    public render(): JSX.Element {
        return (
            <div>
                <Form className='mt-4 bg-white text-center border p-4'>
                    <h3>Change Email</h3>
                    <Input
                        className='form-control'
                        value={this.props.email}
                        onChange={this.onEmailChange}

                    />
                    <Button
                        className='btn mt-3'
                        color='danger'
                        onClick={this.changeEmail}
                    >
                        <i className='fa fa-edit pr-3'/>
                        Change
                    </Button>
                </Form>
            </div>

        );
    }
}
