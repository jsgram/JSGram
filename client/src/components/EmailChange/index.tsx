import React from 'react';
import { connect } from 'react-redux';
import { changeEmail, setEmailText } from '../../store/emailChange/actions';
import { Button, Form, Input, FormGroup, Label, Col } from 'reactstrap';
import { IUser } from '../../store/commonInterfaces/commonInterfaces';

interface IStateToProps {
    setEmailText: any;
    email: string;
    user: IUser;
    changeEmail: any;
}

class Index extends React.Component <IStateToProps> {

    public onEmailChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
        this.props.setEmailText(event.target.value);
    }

    public changeEmail = (): void => {
        this.props.changeEmail(this.props.user, this.props.email);
    }

    public render(): JSX.Element {
        return (
            <div>
                <h3 className='text-center font-weight-light text-secondary text-uppercase'>Change Email</h3>
                <Form className='d-flex flex-column mt-3 bg-white p-4'>
                    <FormGroup row className='align-items-center'>
                        <Label className='col-sm-3 m-0'>
                            Change Email
                        </Label>
                        <Col className='col-sm-9'>
                            <Input
                                className='form-control'
                                value={this.props.email}
                                onChange={this.onEmailChange}
                            />
                        </Col>
                    </FormGroup>
                    <Button
                        className='align-self-center btn mt-3'
                        color='danger'
                        onClick={this.changeEmail}
                    >
                        <i className='fa fa-edit pr-3'/>
                        Change Email
                    </Button>
                </Form>
            </div>
        );
    }
}

const mapStateToProps = (state: any): {email: string, user: IUser} => ({
    email: state.changeEmail.email,
    user: state.profile.user,
});

const mapDispatchToProps = {
    setEmailText,
    changeEmail,
};

export default connect(mapStateToProps, mapDispatchToProps)(Index);
