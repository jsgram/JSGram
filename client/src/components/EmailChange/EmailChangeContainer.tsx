import React from 'react';
import { connect } from 'react-redux';
import EmailChange from '../EmailChange/EmailChange';
import { changeEmail, setEmailText } from '../../store/emailChange/actions';

interface IStateToProps {
    setEmailText: any;
    email: string;
    changeEmail: any;
}

class EmailChangeContainer extends React.Component <IStateToProps> {

    public render(): JSX.Element {
        return (
            <div className='container'>
                <EmailChange
                    setEmailText={this.props.setEmailText}
                    email={this.props.email}
                    changeEmail={this.props.changeEmail}
                />
            </div>
        );
    }
}

const mapStateToProps = (state: any): {email: string} => ({
    email: state.changeEmail.email,
});

const mapDispatchToProps = {
    setEmailText,
    changeEmail,
};

export default connect(mapStateToProps, mapDispatchToProps)(EmailChangeContainer);
