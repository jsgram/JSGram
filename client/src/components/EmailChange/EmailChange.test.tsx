import React from 'react';
import EmailChange, { Index } from './index';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

describe('EmailChange smart component', () => {
    let renderer: any;
    const props = {
        user: {
            username: 'some value',
            email: 'some value',
            fullName: 'some value',
            password: 'some value',
            photoPath: 'some value',
            description: 'some value',
        },
        email: 'some value',
        changeEmail: jest.fn(() => 'some value'),
        setEmailText: jest.fn(() => 'some value'),
    };
    const store = configureStore()(props);

    beforeEach(() => {
        renderer = shallow(<Index {...props} />);
    });

    test('render-success', () => {
        renderer = shallow(<Provider store={store}><EmailChange/></Provider>);
        expect(renderer).toMatchSnapshot();
    });
});
