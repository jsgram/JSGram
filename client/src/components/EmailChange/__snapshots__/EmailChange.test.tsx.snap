// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailChange smart component render-success 1`] = `ShallowWrapper {}`;
